src/main/java/com/puppycrawl/tools/checkstyle/DetailAstImpl.java

```diff
@@ -511,11 +510,9 @@ public List<Token> getHiddenBefore() {
      * @return list of comment tokens
      */
     public List<Token> getHiddenAfter() {
```

Comment:
What is the impact of returning an empty list instead of `null` here?

Reply:
I do not know, we can think of this later on in separate PR.

Reply:
there is discussion at https://github.com/hcoles/pitest/issues/1300

Reply:
> What is the impact of returning an empty list instead of null here?

It looks like the only impact here is changing `JavaParser` which is the only caller of this code and has a `null` check. This is probably left over code from the mentality if it isn't "populated" set it to `null`.

https://github.com/search?q=repo%3Acheckstyle%2Fcheckstyle%20getHiddenAfter&type=code

This wouldn't really impact downstream applications if someone was using this public method, since I would assume most programmers would treat `null` and an empty list the same.

Reply:
do you agree to update this in separate issue ?

Reply:
> do you agree to update this in separate issue ?

The title of the issue for this PR is `Remove UnmodifiableCollectionUtil...`, and NOT returning null here would allow us to readily do a simple return. So - why should this be done in a separate issue? In 2024, returning a `null` collection is frowned upon anyway.

Reply:
>since I would assume most programmers would treat null and an empty list the same.

Not sure I agree here, but at least going from `null` to an empty list should be a backwards compatible change (not breaking)

Reply:
Ok, looks like others wants to kill this null now.
@Kevin222004 , please do it.

Reply:
@Kevin222004 please ping me once we are good here

Reply:
@nrmancuso sure.

I have not started working. I will finish this as soon as possible.